### ISSUE TITLE
Federated login in react native requires two login attempts to work only on production ANDROID

### ISSUE THREAD

**Issue by OtayNacef** [aws-amplify/amplify-js#12926]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Authentication

### Amplify Version

v6

### Amplify Categories

auth

### Backend

Other

### Environment information

<details>

```
  npmPackages:
     ...
   @aws-amplify/react-native: 1.0.13 => 1.0.13 
    @aws-amplify/rtn-web-browser: 1.0.13 => 1.0.13 
    aws-amplify: 6.0.13 => 6.0.13 
    aws-amplify/adapter-core:  undefined ()
    aws-amplify/analytics:  undefined ()
    aws-amplify/analytics/kinesis:  undefined ()
    aws-amplify/analytics/kinesis-firehose:  undefined ()
    aws-amplify/analytics/personalize:  undefined ()
    aws-amplify/analytics/pinpoint:  undefined ()
    aws-amplify/api:  undefined ()
    aws-amplify/api/server:  undefined ()
    aws-amplify/auth:  undefined ()
    aws-amplify/auth/cognito:  undefined ()
    aws-amplify/auth/cognito/server:  undefined ()
    aws-amplify/auth/enable-oauth-listener:  undefined ()
    aws-amplify/auth/server:  undefined ()
    aws-amplify/datastore:  undefined ()
    aws-amplify/in-app-messaging:  undefined ()
    aws-amplify/in-app-messaging/pinpoint:  undefined ()
    aws-amplify/push-notifications:  undefined ()
    aws-amplify/push-notifications/pinpoint:  undefined ()
    aws-amplify/storage:  undefined ()
    aws-amplify/storage/s3:  undefined ()
    aws-amplify/storage/s3/server:  undefined ()
    aws-amplify/storage/server:  undefined ()
    aws-amplify/utils:  undefined ()
    axios: 1.6.7 => 1.6.7 
    react: 18.2.0 => 18.2.0 
    react-native: 0.73.3 => 0.73.3 (0.73.2)
    yarn: ^1.22.21 => 1.22.21 

  npmGlobalPackages:
    appium: 2.4.1
    corepack: 0.19.0
    npm: 9.8.1

```

</details>

### Describe the bug

After migrating to Amplify v6 and deploying the app on production or internal test , I've encountered an issue with federated login on Android devices with no error returned .
The problem occurs only during the initial login attempt with Google authentication. The user needs to sign in twice to be successfully authenticated. Interestingly, this issue is not present in non-production environments, and subsequent login attempts work as expected.

### Expected behavior

The expected behavior is that users should be successfully authenticated on the first attempt when signing in with Google

### Reproduction steps

1. Download the app from internal test (Android).
2. Attempt to sign in with Google for the first time.
3. Observe that the authentication process doesn't succeed on the first attempt.
4. Retry the sign-in by clicking again and selecting the Google account.
5. Confirm that the authentication works correctly on the second attempt.

Note: This issue is specific to production environments on Android, and subsequent login attempts after the initial occurrence function as expected.

### Code Snippet

```javascript
useEffect(() => {
Amplify.configure(awsconfig as ResourcesConfig);
    const unsubscribe = Hub.listen('auth', async ({payload}) => {
      switch (payload.event) {
        case 'signInWithRedirect': {
          await getUser();

          break;
        }
        }
    });

    return unsubscribe;
  }, [getUser]);

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by cwomack** [comment]
Hello, @OtayNacef 👋.  Sorry to hearing you're experiencing this issue.  Are you able to share the frontend code that's causing this behavior?  And is this on Android emulators or real, physical devices?

**Comment by OtayNacef** [comment]
Real device  samsung S21

App.tsx 
```
Amplify.configure(awsconfig as ResourcesConfig);
```

```
useEffect(() => {
Amplify.configure(awsconfig as ResourcesConfig);
    const unsubscribe = Hub.listen('auth', async ({payload}) => {
      switch (payload.event) {
        case 'signInWithRedirect': {
          await getUser();

          break;
        }
        }
    });

    return unsubscribe;
  }, [getUser]);
```

**Comment by OtayNacef** [comment]
@cwomack  added a toast inside hub.listen https://github.com/aws-amplify/amplify-js/assets/46994974/0b061812-af41-4221-839c-0aca534f354c

**Comment by OtayNacef** [comment]
The problem can be fixed by making sure Hub.listen is set up right in the same part of your code where you handle logins, especially for handling signInWithRedirect events.

**Comment by OtayNacef** [comment]
@gokaten-token  

    try {
      await signInWithRedirect(signInInput);
      // TODO : TEMP FIX ANDROID BUG Amplify first signin redirect
      if (IS_ANDROID && !__DEV__)
        setTimeout(async () => {
          await signInWithRedirect(signInInput);
        }, 1000);

    } catch (error) {

**Comment by cwomack** [comment]
Updated this issue to be a bug as we have reproduced it on our side and believe it is related to #13443.  Thank you @OtayNacef and @gokaten-token for the additional context!

**Comment by OtayNacef** [comment]
@cwomack Thanks for the update! We'll keep an eye on issue for any progress. 
Appreciate the coordination and effort.

**Comment by ashwinkumar6** [comment]
Hi @OtayNacef @gokaten-token
Thank you for raising this. We were able to reproduce the issue from our side on Android prod.

From our investigation, this is an issue with the way ReactNative's `AppState` API is instantiated on Android in prod. We're also created an [issue](https://github.com/facebook/react-native/issues/45418) on the ReactNative repo to investigate further on this.

In the mean time we're also working on a workaround from our side. Thank you for your patience

**Comment by ashwinkumar6** [comment]
Hey @OtayNacef @gokaten-token 
The fix has been released in [aws-amplify@6.4.3](https://github.com/aws-amplify/amplify-js/releases/tag/aws-amplify%406.4.3). Thanks for raising this issue !
